src/gameLogic/MovementPattern.js

```diff
@@ -0,0 +1,30 @@
+export default class MovementPattern {
+
+  constructor(origin, destination) {
+    this.origin = origin
+    this.destination = destination
+    this.speed = 0.01
+
+    this.currentLocation = origin
+  }
+
+  step() {
+    this.currentLocation = this.peek()
+  }
+
+  peek() {
```

Comment:
It's a somewhat standardized term for looking at the next item in a queue/stack/etc. without popping it ( https://en.wikipedia.org/wiki/Peek_(data_type_operation) ). But maybe some determineNextStep() would be better. Another question is whether we should even have that. It's just based on some vague idea that we need it in the near future.